docs/modules/plugins/pages/plugin-azure-service-bus.adoc

```diff
@@ -13,25 +13,122 @@ include::partial$azure-authentication.adoc[]
+----
+
+=== Stop consume messages
+
+Stops the Azure Service Bus consumer started by the <<_start_consuming_messages>> step before.
+
+All recorded messages are kept and can be drained into the variable using the step <<_save_messages_to_the_variable>>.
+
+[source,gherkin]
+----
+When I stop consuming messages from `$serviceBusKey` service bus
+----
+
+* `$serviceBusKey` - The key associated with the Azure Service Bus client, will be used to find required properties values.
+
+.Stop consuming messages from the topic configured by Service Bus key `myProjectTopic`
+[source,gherkin]
+----
+When I stop consuming messages from `myProjectTopic` service bus
+----
+
+=== Wait for the messages
+
+Waits until the count of the consumed messages (from the consumer start or after the last draining operation) matches to the rule or until the timeout is exceeded.
+
+[source,gherkin]
+----
+When I wait with `$timeout` timeout until count of consumed by `$serviceBusKey` service bus messages is $comparisonRule `$expectedCount`
+----
+
+* `$timeout` - The maximum time to wait for the events in {durations-format-link} format.
+* `$serviceBusKey` - The key associated with the Azure Service Bus client, will be used to find required properties values.
+* `$comparisonRule` - xref:parameters:comparison-rule.adoc[The comparison rule].
+* `$expectedCount` - The expected count of the events to be matched by the rule.
+
+.Wait for at least 5 messages from the queue for 30 seconds
+[source,gherkin]
+----
+When I wait with `PT30S` timeout until count of consumed by `myProjectQueue` service bus messages is >= `5`
+----
+
+=== Save messages to the variable
+
+Drains/Peeks the consumed messages to the specified variable. If the consumer is not stopped, the new events might arrive after the draining.
+If the consumer is stopped, all the events received from the consumer start or after the last draining operation are stored to the variable.
+
+[source,gherkin]
+----
+When I $queueOperation consumed `$serviceBusKey` service bus messages to $scopes variable `$variableName`
+----
+
+* `$queueOperation` - `DRAIN` - saves the messages consumed since the last drain or from the consumption start and moves the consumer cursor to the position after the last consumed message;
+`PEEK` - saves the messages consumed since the last drain or from the consumption start and doesn't change the consumer cursor position.
+* `$serviceBusKey` - The key associated with the Azure Service Bus client, will be used to find required properties values.
+* `$scopes` - xref:commons:variables.adoc#_scopes[The comma-separated set of the variables scopes].
+* `$variableName` - The variable name to store the messages. The messages are accessible via zero-based index, e.g. `${my-var[0]}` will return the first received message.
+
+.Peek messages from the consumer and save to variable
+[source,gherkin]
+----
+When I PEEK consumed `myProjectQueue` service bus messages to STORY variable `queueMessages`
```

Comment:
please add a set of complete examples to get understanding how to use these steps (in the same way we have for Kafka)